### ISSUE TITLE
Kata id becomes nil if every character in the title are of specific characters

### ISSUE THREAD

**Issue by Voileexperiments** [Codewars/codewars.com#947]
So, a thought of curiosity comes to me earlier: there are two types of kata link: object id notation and kata title notation. Kata title notation automatically converts the title into a verbose link, which is pretty smart and will word-ify some symbols like .(dot), +(plus), =(equal), and even chinese characters, but punctuations, dashes and whitespaces are ignored. So it's possible to make a kata with a name nothing but them and create a kata of... id `nil`. And yes, you can publish the kata.

Currently the kata is in here:
http://www.codewars.com/kata/590b93d05902b81f1c0001a3

![image](https://cloud.githubusercontent.com/assets/14909989/25728644/f352dcf6-3162-11e7-8b8c-3542e0a92291.png)

If you try to open it, you'll get an error (it's inaccessible):

![image](https://cloud.githubusercontent.com/assets/14909989/25726587/785c79ae-3157-11e7-96e6-f514bd45764a.png)

There are safeguards in place to avoid that to happen easily (if you change an already saved kata title to those kind of titles afterwards, the kata title notation just doesn't update), but it's possible to do this if it's saved from scratch.

Another noticable thing is that publishing another kata that resolves to the same id (nil) is impossible. As a refresher, if two katas that resolves to the same kata title notation, the later ones will be appended with `-1`, `-2`, etc. However, it it seems that because the id is already `nil`, doing so ends up choking the kata saving/publishing process and hence failes.

**Comment by Voileexperiments** [comment]
Said kata's information obtained via CW API:

http://www.codewars.com/api/v1/code-challenges/590bba7d51ab820001000057

```json
{"id":"590bba7d51ab820001000057","name":"???- ! - ! - ! - ! - ! - ! - ! - ! -???","slug":null,"category":"reference","publishedAt":"2017-05-04T23:34:27Z","approvedAt":null,"languages":["javascript"],"url":"http://www.codewars.com/kata/590bba7d51ab820001000057","rank":{"id":null,"name":null,"color":null},"createdAt":"2017-05-04T23:34:21Z","createdBy":{"username":"Voile","url":"http://www.codewars.com/users/Voile"},"description":"This kata is inaccessible.\n\n(return the title of the kata to pass the test)","totalAttempts":0,"totalCompleted":1,"totalStars":1,"voteScore":0,"tags":["Fundamentals"],"contributorsWanted":true,"unresolved":{"issues":0,"suggestions":0}}
```

**Comment by Voileexperiments** [comment]
I investigated into this again, the publish request failed with status code 500 and a message of

```
E11000 duplicate key error index: codewars-production.code_challenges.$_slugs_1 dup key: { : undefined } (11000)
```

**Comment by Voileexperiments** [comment]
@kazk apparently people can access this kata without problems again. Was something changed when data fix was applied to katas?

**Comment by kazk** [comment]
Yeah, accessing it should be fixed now. The cause was trying to generate URL based on a slug that's `nil` so I just changed to always use the document id instead.

I haven't done anything about the other issues you mentioned.
